src/utils.h

```diff
@@ -377,5 +377,167 @@ Rescale(Eigen::ArrayXXd               weights,
   intercept_archive.emplace_back(std::move(intercept));
 }
 
+
+//' Return a vector of random index
+//' 
+//' @param n_samples the sampler uses range from 0 to n_samples
+//' @param length the number of random index
+
+Eigen::ArrayXXi
+IndexStochastic(const unsigned n_samples,
+                const unsigned length)
+{
+  Eigen::ArrayXXi index(1, length);
+  for (unsigned i = 0; i < length; ++i) {
+    unsigned s_ind = floor(R::runif(0.0, n_samples));
+    index.col(i) = s_ind;
+  }
+  return index;
+}
+
+//' Return a vector of cyclic index
+//' 
+//' @param n_samples the sampler uses range from 0 to n_samples
+//' @param length the number of random index
+
+Eigen::ArrayXXi
+IndexCyclic(const unsigned n_samples,
+            const unsigned length)
+{
+  Eigen::ArrayXXi index(1, length);
+  index.row(0) = Eigen::ArrayXi::LinSpaced(n_samples, 0, n_samples);
```

Comment:
Can't we just return the `LinSpaced` object here? 

I'm also a bit confused on the design - this looks like it will always give the same samples at each iteration.

Reply:
Hi Michal, I changed it to give cyclic samples with random start at each iteration.